bootloader/src/pyi_utils.c

```diff
@@ -257,6 +257,50 @@ pyi_unsetenv(const char *variable)
 
 #ifdef _WIN32
 
+/* Resolve the runtime tmpdir path and build nested directories */
+wchar_t 
+*pyi_build_temp_folder(char *runtime_tmpdir)
+{
+    wchar_t *wruntime_tmpdir;
+    wchar_t wruntime_tmpdir_expanded[PATH_MAX];
+    wchar_t *wruntime_tmpdir_abspath;
+    wchar_t *cursor;
+    wchar_t path_builder[PATH_MAX];
+    DWORD rc;
+    // Expand environment variables like %LOCALAPPDATA%
+    wruntime_tmpdir = pyi_win32_utils_from_utf8(NULL, runtime_tmpdir, 0);
+    if (!wruntime_tmpdir) {
+        VS("LOADER: Failed to convert runtime-tmpdir to a wide string.\n");
```

Comment:
Please log the errors using `FATALERROR()` or `OTHERERROR()` instead of `VS()`, because the latter is no-op on non-debug bootloaders.